kubetest2-gke/deployer/options/build.go

```diff
@@ -17,41 +17,26 @@ limitations under the License.
 package options
 
 import (
-	"fmt"
-
 	"sigs.k8s.io/kubetest2/pkg/build"
 )
 
 type BuildOptions struct {
-	Strategy      string `flag:"~strategy" desc:"Determines the build strategy to use either make or bazel."`
-	StageLocation string `flag:"~stage" desc:"Upload binaries to gs://bucket/ci/job-suffix if set"`
-	RepoRoot      string `flag:"-"`
-	build.Builder
-	build.Stager
+	CommonBuildOptions *build.Options
 }
 
+var _ build.Builder = &BuildOptions{}
+var _ build.Stager = &BuildOptions{}
+
 func (bo *BuildOptions) Validate() error {
-	return bo.implementationFromStrategy()
+	// force extra GCP files to be staged
```

Comment:
nit: if we're always going to force this, IMHO this should be in `Build()` or really anywhere other than here, `Validate` should normally be only reading data, not writing; this fails the principle of least surprise.

Reply:
Done. also added a check to ensure stage is used with build and up since that's required for gke.